### PR TITLE
feat: add dashboard detail info store

### PR DESCRIPTION
**Pull request by WANZARGEN**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [ ] Bug fixes
- [ ] Feature improvement
- [x] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description

Changed DashboardsDetail & DashboardsCustomize pages to use pinia store.
This is previous work for page structure rearrangement that is to avoid recreation the WidgetContainer when moving between the detail page and the customize page.

### CODE REVIEW

**Review by minseolee — APPROVED** [review]
Nice work!!!

File: src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue [hunk 1]
```diff
@@ -176,30 +159,7 @@ const variableState = reactive({
...
-                enabled: result.settings.date_range.enabled,
-                start: result.settings.date_range.start,
-                end: result.settings.date_range.end,
-            },
-            currency: {
-                enabled: result.settings.currency.enabled,
-                value: result.settings.currency?.value ?? CURRENCY.USD,
-            },
-        };
+        await dashboardDetailStore.getDashboardData(props.dashboardId);
```
Comment (yuda110) [inline src/services/dashboards/dashboard-customize/DashboardCustomizePage.vue:162]: praise: wow simple

**Review by yuda110 — APPROVED** [review]
I like it❤️

**Review by sulmoJ — APPROVED** [review]
LGTM!! 🍍 🍍 🍍

**Pull request merged**